src/main/java/uk/gov/pay/ledger/transaction/dao/mapper/TransactionMapper.java

```diff
@@ -44,6 +45,8 @@ public TransactionEntity map(ResultSet rs, StatementContext ctx) throws SQLExcep
     private ZonedDateTime getZonedDateTime(ResultSet rs, String columnLabel) throws SQLException {
         Timestamp timestamp = rs.getTimestamp(columnLabel);
 
-        return timestamp != null ? ZonedDateTime.ofInstant(timestamp.toInstant(), ZoneOffset.UTC) : null;
+        return Optional.ofNullable(timestamp)
```

Comment:
This looks good! Potentially if the Optional was at the level of the method it would benefit by: 
* simplifying the method, it can just return the value 
* having the code calling the method explicitly define how to handle the "doesn't exist" case

If this seems like a useful pattern it could be implemented as part of another PR.

Reply:
As you've mentioned it can be done as separate PR.

Reply:
https://github.com/alphagov/pay-ledger/pull/120